components/dashboard/src/components/PrebuildLogs.tsx

```diff
@@ -85,59 +85,13 @@ export default function PrebuildLogs(props: PrebuildLogsProps) {
             props.onInstanceUpdate(workspaceInstance);
         }
         switch (workspaceInstance?.status.phase) {
-            // unknown indicates an issue within the system in that it cannot determine the actual phase of
-            // a workspace. This phase is usually accompanied by an error.
-            case "unknown":
-                break;
-
             // Preparing means that we haven't actually started the workspace instance just yet, but rather
             // are still preparing for launch. This means we're building the Docker image for the workspace.
             case "preparing":
-                getGitpodService().server.watchWorkspaceImageBuildLogs(workspace!.id);
-                break;
-
-            // Pending means the workspace does not yet consume resources in the cluster, but rather is looking for
-            // some space within the cluster. If for example the cluster needs to scale up to accomodate the
-            // workspace, the workspace will be in Pending state until that happened.
-            case "pending":
-                break;
-
-            // Creating means the workspace is currently being created. That includes downloading the images required
-            // to run the workspace over the network. The time spent in this phase varies widely and depends on the current
-            // network speed, image size and cache states.
-            case "creating":
-                break;
-
-            // Initializing is the phase in which the workspace is executing the appropriate workspace initializer (e.g. Git
-            // clone or backup download). After this phase one can expect the workspace to either be Running or Failed.
-            case "initializing":
-                break;
-
-            // Running means the workspace is able to actively perform work, either by serving a user through Theia,
-            // or as a headless workspace.
-            case "running":
-                break;
-
-            // Interrupted is an exceptional state where the container should be running but is temporarily unavailable.
-            // When in this state, we expect it to become running or stopping anytime soon.
-            case "interrupted":
-                break;
-
-            // Stopping means that the workspace is currently shutting down. It could go to stopped every moment.
-            case "stopping":
-                break;
-
-            // Stopped means the workspace ended regularly because it was shut down.
             case "stopped":
                 getGitpodService().server.watchWorkspaceImageBuildLogs(workspace!.id);
                 break;
         }
-        if (workspaceInstance?.status.conditions.headlessTaskFailed) {
-            setError(new Error(workspaceInstance.status.conditions.headlessTaskFailed));
-        }
-        if (workspaceInstance?.status.conditions.failed) {
-            setError(new Error(workspaceInstance.status.conditions.failed));
-        }
```

Comment:
If we no longer set these errors here (and no longer display them in the logs), I'm worried that we'll no longer display them anywhere.

If we're removing the explicit `conditions.headlessTaskFailed` and `conditions.failed` messages from here, wouldn't it make sense to explicitly include them in the status message? (Currently, that message seems to always be just "Prebuild failed")

https://github.com/gitpod-io/gitpod/blob/9cd88686371b26d727c6807bd4153d4b26a3f12a/components/dashboard/src/projects/Prebuilds.tsx#L464-L478

Reply:
They are already shown in the footer of the Prebuild logs (by the parent) - see screenshot.

Reply:
Ah, my mistake, my code snippet is from the deprecated `PrebuildInstanceStatus`.

I now see that the new `PrebuildStatusDescription` does embed `prebuild.error` in the message:

https://github.com/gitpod-io/gitpod/blob/9cd88686371b26d727c6807bd4153d4b26a3f12a/components/dashboard/src/projects/Prebuilds.tsx#L354-L360

and this error will include `conditions.failed` or `conditions.headlessTaskFailed` when appropriate:

https://github.com/gitpod-io/gitpod/blob/5ea808e7d43464829e183c1f93f95d3efe784c6f/components/ws-manager-bridge/ee/src/bridge.ts#L79-L81

https://github.com/gitpod-io/gitpod/blob/5ea808e7d43464829e183c1f93f95d3efe784c6f/components/ws-manager-bridge/ee/src/bridge.ts#L87-L90

(Also for what it's worth, that repeated `if` doesn't seem to make sense ☝️)